TLSharp.Core/Network/TcpTransport.cs

```diff
@@ -8,15 +7,22 @@ namespace TLSharp.Core.Network
 {
     public class TcpTransport : IDisposable
     {
+        public delegate TcpClient TcpClientHandler(string address, int port);
```

Comment:
Given that this handler would replace entirely the otherwise `tcpClient.Connect()` call, maybe it's better to call it TcpClientConnectionHandler.